Prlimit Unit Test fail on s390x with go1.21.5

<!-- Make sure that you visit our User Guide at https://kubevirt.io/user-guide.
-->

**What happened**:
Unit tests `prlimit_test.go` fails on s390x wiht go1.21

**What you expected to happen**:
Suceed

**How to reproduce it (as minimally and precisely as possible)**:
`make test-unit`

**Additional context**:
Part of the error output:
```
E0430 13:53:06.524168   21816 prlimit.go:155] failed to kill the process; <nil>
•E0430 13:53:06.548591   21816 prlimit.go:177] faker failed output is:
E0430 13:53:06.548621   21816 prlimit.go:178]
E0430 13:53:06.548634   21816 prlimit.go:179]
E0430 13:53:06.548694   21816 prlimit.go:155] failed to kill the process; os: process already finished
••••

Summarizing 1 Failure:
  [FAIL] Process Limits exec [It] command success with real limits
  /root/go/src/kubevirt.io/containerized-data-importer/pkg/system/prlimit_test.go:48

Ran 7 of 7 Specs in 2.828 seconds
FAIL! -- 6 Passed | 1 Failed | 0 Pending | 0 Skipped
--- FAIL: TestPrLimit (2.83s)
=== RUN   TestHelperProcess
--- PASS: TestHelperProcess (0.00s)
FAIL
FAIL	kubevirt.io/containerized-data-importer/pkg/system	2.852s
```

**Workaround**:
- bump go version in the bazel builder to 1.22
- remove `-coverprofile` flag from test command


**Environment**:
- ported bazel builder image used
- go version 1.21.5

--- a/hack/build/docker/builder/Dockerfile
+++ b/hack/build/docker/builder/Dockerfile
@@ -46,7 +46,7 @@ RUN \
 	go install github.com/mattn/goveralls@latest && \
 	go install golang.org/x/lint/golint@latest && \
 	go install github.com/rmohr/go-swagger-utils/swagger-doc@latest && \
-	go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.13.0 && \
+	go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.14.0 && \
 	go install github.com/securego/gosec/v2/cmd/gosec@latest && \
 	rm -rf "${GOPATH}/pkg"
 

--- a/hack/build/docker/builder/Dockerfile
+++ b/hack/build/docker/builder/Dockerfile
@@ -33,7 +33,7 @@ RUN pip3 install --upgrade j2cli operator-courier==2.1.11 && \
 	ln -s /opt/gradle/gradle-6.6/bin/gradle /usr/local/bin/gradle && \
 	rm gradle-6.6-bin.zip
 
-ENV GIMME_GO_VERSION=1.21.5 GOPATH="/go" GO111MODULE="on"
+ENV GIMME_GO_VERSION=1.22.3 GOPATH="/go" GO111MODULE="on"
 
 RUN mkdir -p /gimme && curl -sL https://raw.githubusercontent.com/travis-ci/gimme/master/gimme | HOME=/gimme bash >> /etc/profile.d/gimme.sh
 

